gcc/cp/contracts.cc

```diff
@@ -2307,7 +2307,24 @@ emit_contract_conditions (tree attrs, tree_code code)
 void
 emit_assertion (tree attr)
 {
+  /* Insert a std::observable () epoch marker.  */
```

Comment:
maybe it's worth making a help fn to build and insert one of these since we need it a few times (not a blocker to merging the patch tho, we can pick it up on another pass)